### PR TITLE
[Gekidou] server_url and theme context

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
Add Context for ServerUrl and Theme with Providers, HOC's and Hooks

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: app/context/theme/index.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    };
+}
+
+export function useTheme(): Theme {
+    return React.useContext(ThemeContext);
+}
+
+const enhancedThemeProvider = withObservables([], ({database}: {database: Database}) => ({
+    currentTeamId: database.get(SYSTEM).query(Q.where('id', SYSTEM_IDENTIFIERS.CURRENT_TEAM_ID)).observe(),
+    themes: database.get(PREFERENCE).query(Q.where('category', Preferences.CATEGORY_THEME)).observeWithColumns(['value']),
```
Comment (avinashlng1080) [inline app/context/theme/index.tsx:85]: `observeWithColumns` is used if the column is sorted, right?

**Review by avinashlng1080 — APPROVED** [review]
LGTM; apart from the one comment.

File: app/context/theme/index.tsx (hunk 1 above)
Comment (enahum) [reply app/context/theme/index.tsx:85]: The wdb example is about sorting, but is really about observing for changes in the column as if the value changes.

Comment (avinashlng1080) [reply app/context/theme/index.tsx:85]: Ok thanks

**Pull request merged**